bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandlerFactory.java

```diff
@@ -36,6 +40,17 @@
 public class ThermostatHandlerFactory extends BaseThingHandlerFactory {
 
     private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_OWD5);
+    private final ZoneId timeZone;
+
+    /**
+     * Creates a new factory
+     *
+     * @param httpClientFactory Factory for HttpClient
+     */
+    @Activate
+    public ThermostatHandlerFactory(@Reference TimeZoneProvider timeZoneProvider) {
+        this.timeZone = timeZoneProvider.getTimeZone();
```

Comment:
Since the user can change the timezone in the openHAB during runtime, you shouldn't cache time zone from TimeZoneProvider. So you will need to pass the TimeZoneProvider into the ThermostatHandler instead.

Reply:
I changed this.

Reply:
@cpmeister By the way: Why does my builds always fail?

Reply:
Almost all of the jenkins builds are failing for one reason or another. It mostly has to do with timing dependent unit tests that are failing because of the high load on the server. As long as your project builds and your project's unit tests pass I wouldn't worry about it.